### PR TITLE
Fixed Highlight module dependency error

### LINKED ISSUE #3795: Highlight module dependency issues on Module Details page
### What you were trying to do

Test a new Omega zip that ended up having some funny versions. Noticed some modules in red within the Gameplay Template drop-down as well as on the Advanced Game Setup page. Would have been nice to also see that after pressing the "Module Details" button while highlighting one of the red modules in the list.

### What actually happened

No highlighting at all on the Module Details page, although it outright lists enough to clearly spot an issue. See screenie below.

Find the code that results in coloring bad modules in red, then also apply it to the Module Details page in various places. See annotated screenie below.

See also #3131 for a bigger picture issue on the topic - but this one spot should be doable on its own without too much work, I hope!

### How to reproduce

* Mess with versions on a given module. For instance change Cities to v9.9.9 and modules like Light And Shadow should fail
* See the red highlight on the Advanced Game Setup page
* With Light & Shadow (or whichever affected module) highlighted click the Module Details page. No highlights!

### Screenies

Now you see it

2019-12-07 01_02_15-Terasology _ Alpha

Now you don't!

2019-12-07 00_53_18-Terasology _ Alpha

To clarify: The minimum required version is 0.6.2, max supported is 0.7.0, yet actual installed version is 1.0.0, which is out of bounds, therefore the dependency resolution fails.

### PR DESCRIPTION
**Pull request by agent-q1**
### Contains

Fixes #3795 . Except, for the text, that doesn't appear red yet

### How to test

Head over to the modules.txt file in any module and change the installed version in it. Whenever the dependency resolution fails, the module appears read in the advancedgamesetup screen. Click on Module details button and you will find that it will remain highlighted red in the module details page as well.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/moduleDetailsScreen/ModuleDetailsScreen.java [hunk 1]
```diff
@@ -187,7 +190,28 @@ public String getString(Module value) {
                     return value.getMetadata().getDisplayName().toString();
                 }
                 return "";
+
+            }
+            @Override
+            public void draw(Module value, Canvas canvas) {
+                 if(!validateModuleDependencies(value.getId()))
+                {
```
Comment (e-aakash) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/moduleDetailsScreen/ModuleDetailsScreen.java]: Formatting of braces seems a bit off.

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/moduleDetailsScreen/ModuleDetailsScreen.java [hunk 2]
```diff
@@ -235,6 +259,7 @@ public String get() {
             }
         });
 
+
```
Comment (e-aakash) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/moduleDetailsScreen/ModuleDetailsScreen.java]: Unnecessary newline here.

File: engine/src/main/java/org/terasology/rendering/nui/widgets/UILabel.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 import org.terasology.math.geom.Vector2i;
 import org.terasology.rendering.assets.font.Font;
-import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.Canvas;	import org.terasology.rendering.nui.*;
```
Comment (e-aakash) [inline engine/src/main/java/org/terasology/rendering/nui/widgets/UILabel.java]: Unused import (+ avoid * imports even if required)

File: config/gradle/common.gradle [hunk 4]
```diff
@@ -84,8 +93,9 @@ test {
     }
 
     jacoco {
-        append = false
-        excludes = ["org.terasology.protobuf.*"]
+       // append = false
```
Comment (jdrueckert) [inline config/gradle/common.gradle]: is this still required? if not, please remove, else it should not be commented out

File: engine/build.gradle [hunk 5]
```diff
@@ -162,43 +160,52 @@ dependencies {
...
     dependsOn classes
 
     doLast {
         // Without the .mkdirs() we might hit a scenario where the classes dir doesn't exist yet
-        file(sourceSets.main.output.classesDir.toString()).mkdirs()
         Reflections reflections = new org.reflections.Reflections(new org.reflections.util.ConfigurationBuilder()
-            .addUrls(configurations.compile.collect { it.toURI().toURL()} + sourceSets.main.output.classesDir.toURI().toURL())
-            .setScanners(new TypeAnnotationsScanner(), new SubTypesScanner()))
-        reflections.save(sourceSets.main.output.classesDir.toString() + "/reflections.cache")
+                .addUrls(inputs.files.collect { it.toURI().toURL() })
```
Comment (jdrueckert) [inline engine/build.gradle]: please recheck the indentation here. looks like it accidentally move one tab in

File: engine/build.gradle [hunk 6]
```diff
@@ -231,24 +238,33 @@ task createVersionInfoFile {
...
+    from sourceSets.main.output.resourcesDir
+    into sourceSets.main.output.classesDirs.first()
+}
+
+jar.dependsOn copyResourcesToClasses
+
 jar.dependsOn createVersionInfoFile
 jar.dependsOn cacheReflections
-jar.finalizedBy cleanReflections
+//jar.finalizedBy cleanReflections
```
Comment (jdrueckert) [inline engine/build.gradle]: is this still required? if not, please remove, else it shouldn't be commented out

File: facades/PC/build.gradle [hunk 7]
```diff
@@ -38,9 +38,10 @@ println "PC VERSION: $version"
 group = 'org.terasology.facades'
 
 dependencies {
-    compile project(':engine')
+    implementation project(':engine')
+    implementation group: 'org.reflections', name: 'reflections', version: '0.9.10'
 
-    /* TODO: Consider whether we can move the CR dependency back here from the engine, where it is referenced from the main menu
+    /// TODO: Consider whether we can move the CR dependency back here from the engine, where it is referenced from the main menu
```
Comment (jdrueckert) [inline facades/PC/build.gradle]: I believe two `/` should suffice 😅

**Pull request merged**